proto/build.gradle.kts

```diff
@@ -0,0 +1,31 @@
+import com.google.protobuf.gradle.*
+
+plugins {
+  id("otel.java-conventions")
+  id("com.google.protobuf")
+}
+
+val protobufVersion = "3.19.4"
+
+dependencies {
+  api("com.google.protobuf:protobuf-java:$protobufVersion")
+}
+
+protobuf {
+  protoc {
+    artifact = "com.google.protobuf:protoc:$protobufVersion"
+  }
+}
+
+sourceSets {
+  main {
+    proto {
+      srcDir("../opamp-spec/proto")
```

Comment:
this is a good question. This is what is suggested in the repo, but I can change it do behave like opentelemetry-java-protos. Perhaps @tigrannajaryan has an opinion here.

Reply:
I like git submodule, unless there are any other advantages w.r.t. task to download the protos.

Reply:
> this is a good question. This is what is suggested in the repo, but I can change it do behave like opentelemetry-java-protos. Perhaps @tigrannajaryan has an opinion here.

@jlegoff I used the submodule in opamp-go since it was the simplest way I knew to bring it as a dependency pinned to a specific commit. If using a gradle download task has any advantages feel free to do that instead. What do you gain by doing that?

Reply:
Honestly, I don't know what we gain by using a download tasks instead of a submodule. Seems like a matter of taste. @jack-berg do you have a strong opinion against using submodules?

Reply:
Nope, no strong opinion. Agree it feels like a matter of taste.